mono/mini/debugger-agent.c

```diff
@@ -5167,7 +5167,11 @@ resume_from_signal_handler (void *sigctx, void *func)
 #endif
 	mono_monoctx_to_sigctx (&ctx, sigctx);
 
-#ifdef PPC_USES_FUNCTION_DESCRIPTOR
+#if defined(PPC_USES_FUNCTION_DESCRIPTOR) && !defined(MONO_ARCH_HAVE_SETUP_RESUME_FROM_SIGNAL_HANDLER_CTX)
```

Comment:
is this ever true?

I think that can be removed, including `mono_ppc_set_func_into_sigctx`

Reply:
I assume it's a theoretical case.... but you're right I guess; anyone porting Mono to a new OS on PPC can just make sure it runs fine with `MONO_ARCH_HAVE_SETUP_RESUME_FROM_SIGNAL_HANDLER_CTX)`.

Still unsure about the Mac OS case, but I don't think Mac OS uses function descriptors in any case.

Reply:
I think it's safe to define `MONO_ARCH_HAVE_SETUP_RESUME_FROM_SIGNAL_HANDLER_CTX` for mac OS too.